students/charlie_england/lesson02/assignment/src/charges_calc.py

```diff
@@ -0,0 +1,138 @@
+        try:
+            data = json.load(file)
+            #Json decode error, expecting property name enclosed in double quotes
+        except Exception as exception:
+            logging.critical(f'Unable to load json file with name {filename} due to {exception}')
+    return data
+
+def calculate_additional_fields(data):
+    """
+    Takes data and iterates through the keys/values
+    series of try, except blocks that will log an error if something occurs
+    converts rental start/end to datetime
+    if the total days is negative: it will switch the start/end days and recalculate total days logging this only if level 3 (debug logging level)
+    
+    Generally the switch in the rental start/end would be a warning level debug for me, 
+    for this exercise I made it a debug level due to there being so many of them
+    and it clogging up the log. I would write a program to fix the data in the source.json if I was to continue
+    """
+    for key, value in data.items():
+        try:
+            rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+            rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+        except Exception as exception:
+            logging.warning(f'FAILED - rental start/rental end {key} with exception {exception}')
+        try:
+            value['total_days'] = (rental_end - rental_start).days
+            if value['total_days'] < 0:
+                logging.debug(f'total days is negative, switching rental start and end for {key}')
+                foo = rental_start
+                rental_start = rental_end
+                rental_end = foo
+                value['total_days'] = (rental_end - rental_start).days
+        except Exception as exception:
+            logging.warning(f'could not update total days for {key} with exception {exception}')
+        try:
+            value['total_price'] = value['total_days'] * value['price_per_day']
+            logging.debug(f'total price for {key} is {value["total_price"]}')
+        except Exception as exception:
+            logging.warning(f"FAILED - total price {key} with exception {exception}")
+        try:
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+        except Exception as exception:
+            logging.warning(f"FAILED - sqrt total price - {key} with exception {exception}")
+        try:
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+        except Exception as exception:
+            logging.warning(f"FAILED - unit cost {key} with exception {exception}")
+    return data
+
+def save_to_json(filename, comp_data):
+    """
+    opens a json file (overwrites the previous one)
+    dumps the completed data into this json file
+    """
+    if '.json' not in filename:
+        filename = f'{filename}.json'
+    with open(filename, 'w') as file:
+        json.dump(comp_data, file)
+
+
+def setup_logger(level='0'):
```

Comment:
Setting the parameter default to 0 here is a great way to default the logging! Though it is probably considered a better option to fix it in the argparse itself. Otherwise you will have to set the default if the arg.debug is ever used anywhere else.